libz-rs-sys/src/gz.rs

```diff
@@ -133,7 +137,12 @@ unsafe fn gzopen_help(path: *const c_char, fd: c_int, mode: *const c_char) -> gz
                 b'a' => state.mode = GzMode::GZ_APPEND,
                 b'+' => {
                     // Read+Write mode isn't supported
-                    free_state(state);
+                    // Safety: we know state is a valid pointer because it was allocated earlier
+                    // in this function, and it is not used after the free because we return
+                    // immediately afterward.
+                    unsafe {
+                        free_state(state);
+                    }
```

Comment:
if you move the `;` to the after of the closing bracket of the unsafe block, this will format on one line, which I personally prefer.